mlflow/sagemaker/__init__.py

```diff
@@ -840,3 +920,115 @@ def _delete_sagemaker_model(model_name, sage_client, s3_client):
     sage_client.delete_model(ModelName=model_name)
 
     return model_arn
+
+
+def _delete_sagemaker_endpoint_configuration(endpoint_config_name, sage_client):
+    """
+    :param sage_client: A boto3 client for SageMaker.
+    :return: ARN of the deleted endpoint configuration.
+    """
+    endpoint_config_info = sage_client.describe_endpoint_config(
+            EndpointConfigName=endpoint_config_name)
+    sage_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    return endpoint_config_info["EndpointConfigArn"]
+
+
+def _find_endpoint(endpoint_name, sage_client):
+    """
+    Finds a SageMaker endpoint with the specified name in the caller's AWS account, returning a
+    NoneType if the endpoint is not found.
+
+    :param sage_client: A boto3 client for SageMaker.
+    :return: If the endpoint exists, a dictionary of endpoint attributes. If the endpoint does not
+             exist, `None`.
+    """
+    endpoints_page = sage_client.list_endpoints(
+        MaxResults=100, NameContains=endpoint_name)
+
+    while True:
+        for endpoint in endpoints_page["Endpoints"]:
+            if endpoint["EndpointName"] == endpoint_name:
+                return endpoint
+
+        if "NextToken" in endpoints_page:
+            endpoints_page = sage_client.list_endpoints(MaxResults=100,
+                                                        NextToken=endpoints_page["NextToken"],
+                                                        NameContains=endpoint_name)
+        else:
+            return None
+
+
+class _SageMakerOperation:
+
+    def __init__(self, status_check_fn, cleanup_fn):
+        self.status_check_fn = status_check_fn
+        self.cleanup_fn = cleanup_fn
+        self.start_time = time.time()
+        self.status = _SageMakerOperationStatus(_SageMakerOperationStatus.STATE_IN_PROGRESS, None)
+        self.cleaned_up = False
+
+    def await_completion(self, timeout_seconds):
+        iteration = 0
+        begin = time.time()
+        while (time.time() - begin) < timeout_seconds:
+            status = self.status_check_fn()
+            if status.state == _SageMakerOperationStatus.STATE_IN_PROGRESS:
+                if iteration % 2 == 0:
+                    # Log the progress status roughly every 10 seconds
```

Comment:
Let's downgrade to every 30 seconds, this gives several screen-fulls of logs right now to do a deployment.

Reply:
Downgraded to 20 seconds after offline discussion about my impatience w.r.t longrunning operations.